lib/GLPI/Agent/Task/Inventory/Generic/Remote_Mgmt/TeamViewer.pm

```diff
@@ -64,20 +74,28 @@ sub _getID {
 
         GLPI::Agent::Tools::Win32->use();
 
-        my $clientid = getRegistryValue(
-            path   => is64bit() ?
-                "HKEY_LOCAL_MACHINE/SOFTWARE/Wow6432Node/TeamViewer/ClientID" :
-                "HKEY_LOCAL_MACHINE/Software/TeamViewer/ClientID",
+        my $clientid = getRegistryKey(                
+                path => "HKEY_LOCAL_MACHINE/SOFTWARE/TeamViewer",            
         );
+        if(!$clientid){
+            my $key = getRegistryKey(
+                path => "HKEY_LOCAL_MACHINE/SOFTWARE/Wow6432Node/TeamViewer",
+            );
+        }  
```

Comment:
Why changing `getRegistryValue()` call to `getRegistryKey()` call here ? If I remember well the purpose of this code was to directly catch `ClientID` on well-known place in registry (probably for older teamviewer versions). I'm not definitively sure but you probably break the original logic here. This is also the reason why variable was named `$clientid`.
Also checking on `Wow6432Node` make no sens if the platform is not 64 bits.
```suggestion
        my $clientid = getRegistryKeyValue(                
                path => "HKEY_LOCAL_MACHINE/SOFTWARE/TeamViewer/ClientID",            
        );
        if (!$clientid && is64bit()) {
            my $key = getRegistryKey(
                path => "HKEY_LOCAL_MACHINE/SOFTWARE/Wow6432Node/TeamViewer/ClientID",
            );
        }  
```